docs/faq.md

```diff
@@ -142,3 +142,23 @@ Refer to the [PSPDFKit blog post for more details.](https://pspdfkit.com/blog/20
 ```swift
 UIApplication.sharedApplication().keyWindow?.layer.speed = 100
 ```
+
+**How do I reset application state before each test using Swift?**
+
+In the application's Build Settings, set **Defines Module** to **Yes**. Create a `resetApplication()` method on the AppDelegate.
+
+In the EarlGrey test target, import the application using `@testable`. In `setUp()`, aquire a reference to the delegate then invoke `resetApplication()`.
+
+```swift
+// swift
+@testable import App
+
+class MyTests: XCTestCase {
+
+    override func setUp() {
```

Comment:
> Are you worried about the app crashing or the same device previously having the app installed?
> In a clean simulator scenario, when a test is run for the first time, shouldn't the app be in a clean state?

I'm worried about the app already being on the simulator / used for manual testing. We have no guarantees about the "clean simulator scenario" when doing EarlGrey testing.